gradle/dependencies.gradle

```diff
@@ -179,10 +183,10 @@ libs += [
   powermockEasymock: "org.powermock:powermock-api-easymock:$versions.powermock",
   reflections: "org.reflections:reflections:$versions.reflections",
   rocksDBJni: "org.rocksdb:rocksdbjni:$versions.rocksDB",
-  scalaCollectionCompat: "org.scala-lang.modules:scala-collection-compat_$versions.baseScala:$versions.scalaCollectionCompat",
-  scalaJava8Compat: "org.scala-lang.modules:scala-java8-compat_$versions.baseScala:$versions.scalaJava8Compat",
   scalaLibrary: "org.scala-lang:scala-library:$versions.scala",
-  scalaLogging: "com.typesafe.scala-logging:scala-logging_$versions.baseScala:$versions.scalaLogging",
+  scala3Library: "org.scala-lang:scala3-library_3:3.0.0",
```

Comment:
Why can we not set scalaLibrary with the correct name depending on the version? That would make a lot of changes in `build.gradle` unnecessary.

Reply:
I can add a couple of extra variables, but please notice that the library changed in the name as well:
it's `scala3-library_3:3.0.0` vs. `scala-library_2.13:2:13.16
Notice the extra `3` right after `scala`.

Reply:
I know, but I don't see the value of exposing that outside of this file. We only use one scalaLibrary for a given build.

Reply:
I had something like this on my early attempts, but I went for this as it was "clearer" for the reviewer. I go back to that